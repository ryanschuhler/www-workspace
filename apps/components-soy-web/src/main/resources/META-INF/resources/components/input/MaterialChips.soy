{namespace MaterialChips}

/**
 * MaterialChips component
 */
{template .render}
  {@param chips: list<string>}
  {@param? inputId: string}
  {@param? inputName: string}
  {@param? maxChipLength: number}
  {@param? placeholder: string}
  {@param? selectedIndex: number}
  {@param? readonly: bool}

  {@param? onChipDeleteClicked_: any}

  {if $readonly}
    <div class="chips">
      <span>
        {foreach $chip in $chips}
          {if $chip != ''}
            {let $index: index($chip) /}

            {call MaterialChip.render}
              {param events: ['chipDeleteClicked': $onChipDeleteClicked_] /}
              {param deletable: not $readonly /}
              {param elementClasses: $index == $selectedIndex ? 'selected' : '' /}
              {param label: $chip /}
            {/call}
          {/if}
        {/foreach}
      </span>
    </div>
  {else}
    <div class="chips chips-editable">

    <span>
      {foreach $chip in $chips}
        {if $chip != ''}
          {let $index: index($chip) /}

          {call MaterialChip.render}
            {param events: ['chipDeleteClicked': $onChipDeleteClicked_] /}
            {param deletable: true /}
            {param elementClasses: $index == $selectedIndex ? 'selected' : '' /}
            {param label: $chip /}
          {/call}
        {/if}
      {/foreach}
    </span>

      {let $fakeLinkAttributes kind="attributes"}
        href=""
        onBlur="unselectChip_"
        onKeydown="onFakeLinkKeyDown_"
        onKeyup="onFakeLinkKeyUp_"
        ref="anchor"
        tabindex="-1"
      {/let}

      <a {$fakeLinkAttributes}></a>

      {let $inputAttributes kind="attributes"}
        class="chips-input"
        onChange="onInputChange_"
        onInput="onInputChange_"
        onKeydown="onInputKeyDown_"
        onKeypress="onInputKeyPress_"
        onKeyup="onInputKeyUp_"
        onMouseup="onInputChange_"
        onPaste="onInputChange_"
        maxlength="{$maxChipLength}"

        {if $placeholder and length($chips) == 0}
          placeholder="{$placeholder}"
        {/if}

        ref="input"
        type="text"
      {/let}

      <input {$inputAttributes} />

      {if $inputId or $inputName}
        <input id="{$inputId}" name="{$inputName}" type="hidden" value="{$chips}"/>
      {/if}
    </div>

  {/if}

{/template}


