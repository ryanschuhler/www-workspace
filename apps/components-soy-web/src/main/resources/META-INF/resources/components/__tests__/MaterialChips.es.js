'use strict';

import MaterialChips from '../input/MaterialChips.es';

let component;

const chips = [
  'chip1',
  'chip2',
  'chip3'
]

describe('MaterialChips', () => {
  afterEach(() => {
    if (component) {
      component.dispose();
    }
  });

  it('should render', () => {
    component = new MaterialChips({
      chips: chips
    });

    expect(component).toMatchSnapshot();
  });
});