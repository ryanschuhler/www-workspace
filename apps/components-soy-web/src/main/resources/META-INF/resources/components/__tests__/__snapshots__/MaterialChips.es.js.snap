// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaterialChips should render 1`] = `
<div class="chips chips-editable">
  <span>
    <span class="mdl-chip mdl-chip--deletable">
      <span class="mdl-chip__text">chip1</span>
      <button type="button" class="mdl-chip__action">
        <i class="icon-root pario-icon lexicon-icon material-icons">cancel</i>
      </button>
    </span>
    <span class="mdl-chip mdl-chip--deletable">
      <span class="mdl-chip__text">chip2</span>
      <button type="button" class="mdl-chip__action">
        <i class="icon-root pario-icon lexicon-icon material-icons">cancel</i>
      </button>
    </span>
    <span class="mdl-chip mdl-chip--deletable">
      <span class="mdl-chip__text">chip3</span>
      <button type="button" class="mdl-chip__action">
        <i class="icon-root pario-icon lexicon-icon material-icons">cancel</i>
      </button>
    </span>
  </span>
  <a href="" ref="anchor" tabindex="-1"></a>
  <input class="chips-input" ref="input" type="text">
</div>
`;
